### PR TITLE
Return zero gradients for structurally independent variables

### PR DESCRIPTION
**Pull request by alanlujan91**
Follows up on #129, where the open question was the return contract of the gradient utilities. The utility `compute_gradients_for_tensors` now returns a zero tensor when a `wrt` variable has no computational path to the target, since zero is the value of the partial derivative under structural independence; passing a tensor without `requires_grad=True` raises `ValueError` instead of silently producing `None`. Under `create_graph=True` the materialized zeros stay graph-connected, so higher-order differentiation through them still works. The `BellmanPeriod` gradient methods inherit the tensor-only contract.

Two consumer updates follow from this. The Euler and FOC diagnostics now raise when the marginal reward is zero at every sample point, and the chain-rule return factor skips arrival states whose transition gradient is all zero, so a non-finite pre-state factor cannot turn `0 * Inf` into `NaN` where the old `None` path used to skip the term.

New regression tests pin the composed-reward gradient against analytic values (the substantive guarantee from #129), second-order differentiation through a structural zero, and the `NaN`-masking path. The PR also declares `torch >=2.0` as the minimum supported PyTorch version.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import torch
 from skagent.utils import compute_gradients_for_tensors
 
+torch.manual_seed(10077696)
+
```
[the remark this answers is not in the document]
Comment (sbenthall) [reply tests/test_utils.py]: @alanlujan91 I think it makes sense to move this seed
